File: 1688. Count of Matches in Tournament/soln1.py
# Runtime
# 23ms
# Beats 99.48%of users with Python3

# Memory
# 16.25MB
# Beats 35.39%of users with Python3

class Solution:
    def numberOfMatches(self, n: int) -> int:
        matches = 0
        while n > 1:
            if n % 2 == 1:
                n = int(n / 2)
                matches += n
                n += 1
            else:
                n = int(n / 2)
                matches += n

        return matches

File: 49. Group Anagrams/soln1.cpp
// Runtime
// 55ms
// Beats 17.64% of users with C++

// Memory
// 31.32MB
// Beats 6.77% of users with C++

class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<vector<int>,vector<int>> profiles;
        
        for (int i = 0; i < strs.size(); i++){
            vector<int> profile (26,0);
            for (int j = 0; j < strs[i].size(); j++){ // build string profile
                profile[strs[i][j]-'a']++;
            }
            if  (profiles.count(profile) == 0) profiles[profile]= {i};// add it to grouping
            else profiles[profile].push_back(i);

        }
        vector<vector<string>> v;
        map<vector<int>, vector<int>>::iterator it = profiles.begin(); 
        
        while (it != profiles.end()){ // take mapped values and put them in returned vector
            vector<string> stArr;
            vector<int> indices;
            indices = it->second;
            for (int i = 0; i < indices.size(); i++){
                stArr.push_back(strs[indices[i]]);
            }
            v.push_back(stArr);
            it++;
        }
        return v;
    }
};
